### DISCUSSION TITLE
Add Aws SDK Kotlin Support

### DISCUSSION THREAD

**Discussion by hantsy** [awspring/spring-cloud-aws#256] [Ideas]
Awslabs released a Kotlin variant for AWS SDK, add Kotlin and Kotlin Coroutines support.

**Comment by maciejwalkowiak** [comment]
Elaborate please on how do you think this integration could work/look like?

**Comment by driverpt** [comment]
@hantsy , AWS SDK for Kotlin is targeted at Kotlin Native Projects. You can just bridge `SdkAsyncClient` to Co-routines

**Comment by hantsy** [comment]
@driverpt 

>AWS SDK for Kotlin is targeted at Kotlin Native Projects.

It is just Kotlin variant for the original Java version, provides a lot of extensions for Kotlin DSL and Coroutines to simplify the usage in a Kotlin project, see the [YouTube vedio](https://www.youtube.com/watch?v=1o1G3QIkipI)

@maciejwalkowiak For developers what we need is like the Kotlin support in Spring, make all AWS SDK Kotlin extensions available in this umbrella project. I think it is just an extra kotlin specific module.

**Reply by maciejwalkowiak** [reply]
I've converted this into discussion. Once we come up something actionable on how this support can look like, we can set up an issue.

@hantsy at this stage i am not sure how this support could look like. We could definitely make Spring Cloud AWS more Kotlin friendly (like explicit nullability annotations - what i want to add in v3). Perhaps we can auto-configure Kotlin clients from AWS SDK? Not sure if it is worth the effort. I am very open for your ideas - you are welcome to sketch a draft PR.

**Comment by driverpt** [comment]
It has support for JVM Platform, but the purpose of AWS Kotlin SDK is to support all Kotlin MPP Target Runtimes

**Comment by Jarvvski** [comment]
pretty old, but was there any movement on this @maciejwalkowiak? 

The kotlin SDK is pretty well maintained by AWS - https://github.com/aws/aws-sdk-kotlin
